nexus/tests/integration_tests/instances.rs

```diff
@@ -1081,6 +1085,388 @@ async fn test_instance_migrate_v2p_and_routes(
+        .unwrap()
+        .expect("migrated instance should still have a sled")
+        .sled_id;
+
+    assert_eq!(current_sled, dst_sled_id);
+
+    let migration = dbg!(migration_fetch(cptestctx, migration_id).await);
+    assert_eq!(migration.target_state, MigrationState::Completed.into());
+    assert_eq!(migration.source_state, MigrationState::Completed.into());
+}
+
+// An instance that requires a Turin CPU will be placed on the Turin sled, and
+// if we're told to migrate it to a Milan sled, we'll error instead.
+#[nexus_test]
+async fn test_instance_migration_incompatible_cpu_platforms(
+    cptestctx: &ControlPlaneTestContext,
+) {
+    let client = &cptestctx.external_client;
+    let internal_client = &cptestctx.internal_client;
+    let apictx = &cptestctx.server.server_context();
+    let nexus = &apictx.nexus;
+    let instance_name = "bird-ecology";
+
+    // Set up a second sled-agent representing a sled with a Turin processor.
+    // The instance will require Turin, so it will be placed here.
+    let nexus_address =
+        cptestctx.server.get_http_server_internal_address().await;
+
+    let config = omicron_sled_agent::sim::Config::for_testing(
+        SledUuid::new_v4(),
+        omicron_sled_agent::sim::SimMode::Explicit,
+        Some(nexus_address),
+        Some(&camino::Utf8Path::new("/an/unused/update/directory")),
+        omicron_sled_agent::sim::ZpoolConfig::None,
+        nexus_sled_agent_shared::inventory::SledCpuFamily::AmdTurin,
+    );
+    let turin_sled_id = config.id;
+
+    let _turin_sled = start_sled_and_wait(cptestctx, config).await;
+
+    let milan_sled_id = cptestctx.first_sled_id();
+
+    create_project_and_pool(&client).await;
+    let instance_url = get_instance_url(instance_name);
+
+    // Explicitly create an instance with no disks. Simulated sled agent assumes
+    // that disks are co-located with their instances.
+    let instance = nexus_test_utils::resource_helpers::create_instance_with(
+        client,
+        PROJECT_NAME,
+        instance_name,
+        &params::InstanceNetworkInterfaceAttachment::Default,
+        Vec::<params::InstanceDiskAttachment>::new(),
+        Vec::<params::ExternalIpCreate>::new(),
+        true,
+        Default::default(),
+        Some(InstanceCpuPlatform::AmdTurin),
+    )
+    .await;
+    let instance_id = InstanceUuid::from_untyped_uuid(instance.identity.id);
+
+    // Poke the instance into an active state.
+    instance_simulate(nexus, &instance_id).await;
+    let instance_next = instance_get(&client, &instance_url).await;
+    assert_eq!(instance_next.runtime.run_state, InstanceState::Running);
+
+    let sled_info = nexus
+        .active_instance_info(&instance_id, None)
+        .await
+        .unwrap()
+        .expect("running instance should have a sled");
+
+    // If the Turin-requiring instance isn't on the Turin sled, either the
+    // default simulated sled is now Turin-compatible or something is very
+    // wrong.
```

Comment:
could we perhaps construct a default simulated sled and assert it isn't Turin? so that we know if the test failed for that reason?

Reply:
that'd involve either skipping `nexus_test` here (really don't want to do that for this test, it'll definitely end up with weird test environment skew that way) or plumbing parameters through `nexus_test` (which i'm lukewarm on if we'll do it in exactly one test.. i'm not sure if its useful anywhere else). so that's how i ended up at the assert to point future-us the right way :(